usersimcrs/simulator/agenda_based/interaction_model.py

```diff
@@ -114,32 +127,21 @@ def intent_distribution(
 
             # Extracts conjoint agent intent and user intent pairs
             # from conversations.
-            for j, utterance_record in enumerate(
-                annotated_conversation["conversation"]
-            ):
+            for j, utterance in enumerate(annotated_conversation.utterances):
                 # Only consider agent intent as keys
-                if (
-                    utterance_record["participant"]
-                    != DialogueParticipant.AGENT.name
-                ):
+                if utterance.participant != DialogueParticipant.AGENT.name:
                     continue
-                intent = Intent(utterance_record["intent"])
+                intent = utterance.intent
                 if intent not in intent_dist:
```

Comment:
Make intent_dist a defaultdict(dict)? (I know that it's legacy code)

Reply:
Good idea it will reduce the number of if conditions. Done.